students/cfarris/lessons/lesson08/assignment/inventory.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env Python3
+
+'''
+lesson 8 work with closures
+
+Pylint is complaining about passing invoice and customer-name in partial.
+From my brief research, this appears to be a bug with pylint. 
+'''
+
+from functools import partial
+import csv
+
+
+
+
+
+def add_furniture(item_code="", item_desc="",
+                  item_monthly_price="",
+                  invoice_file="",
+                  customer_name=""):
+    """
+    add_furniture creates an invoice file (or appends if exists)
+    File name is parameter,the rest of the parameters go into invoice_file
+    """
+    with open(invoice_file, 'a+') as i_file:
+        i_file.write(f'{customer_name}, {item_code}, {item_desc}, {item_monthly_price}\n')
+    print("completed")
+
+
+def single_customer(customer_name, invoice_file):
+    """
+    single_customer returns a function that takes one parameter(rental_items)
+    it returns a function that will iterate through rental_items
+    and add each item to the invoice_file.
+    """
+    def add_cust_rentals_to_invoice(client_items):
+        with open(client_items, 'r') as client_i:
+            csv_reader = csv.reader(client_i, delimiter=',')
+            store_increment = 0
+            add_rentals = partial(add_furniture,
+                                  invoice_file=invoice_file,
+                                  customer_name=customer_name,)
+
+            for row in csv_reader:
+                store_increment = store_increment + 1
+                add_rentals(row[0],
+                            row[1],
+                            row[2])
+
+    return add_cust_rentals_to_invoice
```

Comment:
store_increment is not crucial to your function, and you're not returning it, so I wouldn't include it.
I see why pylint is complaining. You have too many things called invoice file. And, it is true, that in this case they are actually all the same file, but it doesn't have to be that way. You could call the input to single_customer general_invoice or something, then in the partial it would be invoice_file=general_invoice. But, I agree that this is one of those fuzzys that you would decide with the customer so that it is clear for them, since that is most important. If you decide it makes sense for the input to both functions to have the same name, tell pylint to ignore that line.